quiche/include/quiche.h

```diff
@@ -630,6 +629,58 @@ enum quiche_h3_error {
     // The requested operation cannot be served over HTTP/3. Peer should retry
     // over HTTP/1.1.
     QUICHE_H3_ERR_VERSION_FALLBACK = -20,
+
+    // QUIC transport had no more work to do.
+    QUICHE_H3_TRANSPORT_ERR_DONE = -1001,
```

Comment:
Instead of hard-coding the numbers, maybe do `QUICHE_ERR_DONE - 1000` or something like that?

It could even be a macro to avoid potential typos, e.g.

```c
enum quiche_h3_error {
    ...

    #define QUICHE_H3_TRANSPORT_ERR_DEF(err) \
        QUICHE_H3_TRANSPORT_ERR_##err = QUICHE_ERR_##err - 1000

    QUICHE_H3_TRANSPORT_ERR_DEF(DONE),

    QUICHE_H3_TRANSPORT_ERR_DEF(BUFFER_TOO_SHORT),

    ...
};
```

Reply:
On one hand the macro is neat. On the other hand, it seems to push the requirement on applications to do the name expansion themselves i.e. when comparing or switching on a returned error code they still need to type out the name in full. 

So I've gone with the simpler suggestion of avoiding hardcoding. This will mean that if people print the error code value out, they can't easily grep for it in the header. However, the calculation is trivial enough that it shouldn't be too hard to figure out.